DS-Session-3/priorityQueue/virtualCommandPrompt/VirtualCommandPrompt.java

```diff
@@ -0,0 +1,214 @@
+package virtualCommandPrompt;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Scanner;
+
+/**
+ * class to create a virtual command prompt
+ * @author Ishita_Mathur
+ *
+ */
+public class VirtualCommandPrompt
+{
+	static List<Directory> listOfDirectory = new ArrayList<Directory>();
+	static String path = ".\\";
+
+	/**
+	 * method to implement the virtual command prompt
+	 */
+	public static void virtualCommandPrompt()
+	{
+		Directory root = new Directory("R:", new Date());
+		listOfDirectory.add(root);
+		Scanner scanner = new Scanner(System.in);
+		System.out.println("***Welcome to Virtual Command Prompt***");
+		String signature = "R:\\";
+		System.out.print(signature + ">");
+		Directory currentDirectory = root;
+
+		while(true)
+		{
+			String commandLine = scanner.nextLine();
+			String[] commandWords = commandLine.split(" ");
+
+			switch(commandWords[0])
+			{
+
+				//command to create a directory in the current directory
+				case "mkdir": if(commandWords.length >= 2)
+					      {
+					              boolean flag = false;
+						        for(Directory directory : currentDirectory.getListOfSubDirectory())
+						        {
+								
+						        	if (directory.getName().equals(commandWords[1]))

```

Comment:
`commandWords[1]` is used many times in this block, a variable should be created instead of such repetitions.